tests/fuzz/pilot_model_fuzzer_to_move.go

```diff
@@ -0,0 +1,240 @@
+		0: "DENY",
+		1: "AUDIT",
+		2: "CUSTOM",
+	}
+)
+
+// createAuthPolicy is a helper to create a policy
+func createAuthPolicy(f *fuzz.ConsumeFuzzer) (*authpb.AuthorizationPolicy, error) {
+	policy := &authpb.AuthorizationPolicy{}
+	err := f.GenerateStruct(policy)
+	if err != nil {
+		return nil, err
+	}
+	policyType, err := f.GetInt()
+	if err != nil {
+		return nil, err
+	}
+	switch policyTypes[policyType%3] {
+	case "DENY":
+		denyPolicy := proto.Clone(policy).(*authpb.AuthorizationPolicy)
+		denyPolicy.Action = authpb.AuthorizationPolicy_DENY
+		return denyPolicy, nil
+	case "AUDIT":
+		auditPolicy := proto.Clone(policy).(*authpb.AuthorizationPolicy)
+		auditPolicy.Action = authpb.AuthorizationPolicy_AUDIT
+		return auditPolicy, nil
+	case "CUSTOM":
+		customPolicy := proto.Clone(policy).(*authpb.AuthorizationPolicy)
+		customPolicy.Action = authpb.AuthorizationPolicy_CUSTOM
+		return customPolicy, nil
+	}
+	return policy, nil
+}
+
+// AuthorizationPoliciesFuzzer implements a fuzzer
+// for authorization policies
+func AuthorizationPoliciesFuzzer(data []byte) int {
+	initter.Do(initFunc)
+	f := fuzz.NewConsumer(data)
+	configQty, err := f.GetInt()
+	if err != nil {
+		return 0
+	}
+	configQty = configQty % 20
+	configs := make([]config.Config, 0)
+	var wantDeny, wantAudit, wantCustom []AuthorizationPolicy
+	wantDeny = make([]AuthorizationPolicy, 0)
+	wantAudit = make([]AuthorizationPolicy, 0)
+	wantCustom = make([]AuthorizationPolicy, 0)
+	for i := 0; i < configQty; i++ {
+		newConfig := config.Config{}
+		err = f.GenerateStruct(&newConfig)
+		if err != nil {
+			return 0
+		}
+		policy, err := createAuthPolicy(f)
+		if err != nil {
+			return 0
+		}
+		newConfig.Spec = policy
+
+		switch policy.Action {
+		case authpb.AuthorizationPolicy_DENY:
+			wantDeny = append(wantDeny, AuthorizationPolicy{
+				Name:      newConfig.Meta.Name,
+				Namespace: newConfig.Meta.Namespace,
+				Spec:      policy,
+			})
+		case authpb.AuthorizationPolicy_AUDIT:
+			wantAudit = append(wantAudit, AuthorizationPolicy{
+				Name:      newConfig.Meta.Name,
+				Namespace: newConfig.Meta.Namespace,
+				Spec:      policy,
+			})
+		case authpb.AuthorizationPolicy_CUSTOM:
+			wantCustom = append(wantCustom, AuthorizationPolicy{
+				Name:      newConfig.Meta.Name,
+				Namespace: newConfig.Meta.Namespace,
+				Spec:      policy,
+			})
+		}
+		newConfig.Meta.GroupVersionKind = collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind()
+		configs = append(configs, newConfig)
+	}
+	if len(configs) == 0 {
+		return 0
+	}
+	t := &testing.T{}
+	authzPolicies, err := createFakeAuthorizationPoliciesFuzz(configs, t)
+	if err != nil {
+		return 0
+	}
+	if len(authzPolicies.NamespaceToPolicies) == 0 {
+		return 0
+	}
+	ns, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	wl := make(map[string]string)
+	err = f.FuzzMap(&wl)
+	if err != nil {
+		return 0
+	}
+	result := authzPolicies.ListAuthorizationPolicies(ns, []labels.Instance{wl})
```

Comment:
I am not sure what we are trying to catch here? This function seems nearly trivial and can easily get 100% test coverage from unit tests. Is our goal to fuzz 100% of the code base? My expectation is that fuzzing is typically more useful on things like parsing arbitrary inputs, etc.

it seems like we practically re-implement ListAuthorizationPolicies in the test code and check the output is the same.

Reply:
Removed that fuzzer.